packages/patterns/src/components/partial-syncing-controls.js

```diff
@@ -1,108 +1,106 @@
 
-function PartialSyncingControls( { name, attributes, setAttributes } ) {
-	const syncedAttributes = PARTIAL_SYNCING_SUPPORTED_BLOCKS[ name ];
-	const attributeSources = syncedAttributes.map(
-		( attributeName ) =>
-			attributes.metadata?.bindings?.[ attributeName ]?.source
-	);
-	const isConnectedToOtherSources = attributeSources.every(
-		( source ) => source && source !== 'core/pattern-overrides'
-	);
+function removeBindings( bindings, syncedAttributes ) {
+	let updatedBindings = {};
+	for ( const attributeName of Object.keys( syncedAttributes ) ) {
+		// Omit any pattern override bindings from the `updatedBindings` object.
+		if (
+			bindings?.[ attributeName ]?.source !== 'core/pattern-overrides' &&
+			bindings?.[ attributeName ]?.source !== undefined
+		) {
+			updatedBindings[ attributeName ] = bindings[ attributeName ];
+		}
+	}
+	if ( ! Object.keys( updatedBindings ).length ) {
+		updatedBindings = undefined;
+	}
+	return updatedBindings;
+}
 
-	// Render nothing if all supported attributes are connected to other sources.
-	if ( isConnectedToOtherSources ) {
-		return null;
+function setBindings( bindings, syncedAttributes ) {
+	const updatedBindings = { ...bindings };
+	for ( const attributeName of Object.keys( syncedAttributes ) ) {
+		if ( ! bindings?.[ attributeName ] ) {
+			updatedBindings[ attributeName ] = {
+				source: 'core/pattern-overrides',
+			};
+		}
 	}
+	return updatedBindings;
+}
+
+export default function useSetPatternBindings(
+	{ name, attributes, setAttributes },
+	currentPostType
+) {
+	const metadataName = attributes?.metadata?.name;
+	const prevMetadataName = usePrevious( metadataName ) ?? metadataName;
+	const bindings = attributes?.metadata?.bindings;
+
+	useEffect( () => {
+		// Bindings should only be created when editing a wp_block post type,
+		// and also when there's a change to the user given name for the block.
+		if (
+			currentPostType !== 'wp_block' ||
+			metadataName === prevMetadataName
+		) {
+			return;
+		}
+
+		const syncedAttributes = PARTIAL_SYNCING_SUPPORTED_BLOCKS[ name ];
+		const attributeSources = Object.keys( syncedAttributes ).map(
+			( attributeName ) =>
+				attributes.metadata?.bindings?.[ attributeName ]?.source
+		);
+		const isConnectedToOtherSources = attributeSources.every(
+			( source ) => source && source !== 'core/pattern-overrides'
+		);
 
-	function updateBindings( isChecked ) {
-		let updatedBindings = {
-			...attributes?.metadata?.bindings,
-		};
+		// Avoid overwriting other (e.g. meta) bindings.
+		if ( isConnectedToOtherSources ) {
+			return;
+		}
 
-		if ( ! isChecked ) {
-			for ( const attributeName of syncedAttributes ) {
-				if (
-					updatedBindings[ attributeName ]?.source ===
-					'core/pattern-overrides'
-				) {
-					delete updatedBindings[ attributeName ];
-				}
-			}
-			if ( ! Object.keys( updatedBindings ).length ) {
-				updatedBindings = undefined;
-			}
+		// The user given name for the block was deleted, remove the bindings.
```

Comment:
```suggestion
		// The user-given name for the block was deleted, remove the bindings.
```

Tiniest of tiny nits x 2, so my code spell plugin and grammarly stop decorating my editor and page 😁